.build/Build.CheckPublicApi.cs

```diff
@@ -0,0 +1,27 @@
+using Nuke.Common;
+using Nuke.Common.CI;
+using Nuke.Common.CI.AzurePipelines;
+using Nuke.Common.CI.GitHubActions;
+using Nuke.Common.Execution;
+using Nuke.Common.Tools.DotNet;
+using static Nuke.Common.Tools.DotNet.DotNetTasks;
+using static Helpers;
+
+partial class Build : NukeBuild
+{
+    Target CheckPublicApi => _ => _
+        .DependsOn(Restore)
+        .Executes(() =>
+        {
+            if (!InvokedTargets.Contains(Restore))
+            {
+                DotNetBuildSonarSolution(AllSolutionFile);
+            }
+
+            DotNetBuild(c => c
+                .SetProjectFile(AllSolutionFile)
+                .SetNoRestore(InvokedTargets.Contains(Restore))
+                .SetConfiguration(Configuration)
+                .SetProperty("RequireDocumentationOfPublicApiChanges", true));
+        });
```

Comment:
@michaelstaib I've added a new Nuke target, so the public API check can be executed via
```sh
./build.sh CheckPublicApi
```
All that's left is that there needs to be a PR check that runs when the PR targets main and the tests have successfully completed.

Reply:
@tobias-tengler  This can be found here:
https://github.com/ChilliCream/hotchocolate/blob/main/.devops/azure-pipelines.test-pr-hotchocolate.yml

Reply:
@PascalSenn I've integrated it in the pipeline successfully!
But there's still a weird issue: I intentionally introduced an undocumented change and locally `./build.sh CheckPublicApi` errors out as expected, but in CI it passes 🤔
Not sure what's going on yet, but I'm on it...